plugins/modules/vmware_guest_controller.py

```diff
@@ -302,12 +312,13 @@ def check_ctl_disk_exist(self, ctl_type=None, bus_number=None):
 
         return ctl_specified, disks_attached_exist
 
-    def create_controller(self, ctl_type, bus_number=0):
+    def create_controller(self, ctl_type, bus_number=0, bus_sharing='noSharing'):
```

Comment:
```suggestion
    def create_controller(self, ctl_type, bus_number=0, bus_sharing):
```

The default here isn't needed. Since `bus_sharing` is defined as having a default of `noSharing` in the  `argument_spec`, it shouldn't be possible to call this function _without_ a value.

Reply:
> An argument without default following one with throws an exception when defining a function.
> move it before the `bus_number`?

Ah, yes... I didn't realize this. Of course you're right, `bus_sharing` should be moved before `bus_number`.